$NetBSD: patch-Makefile,v 1.10 2026/03/10 20:40:57 wiz Exp $

- Libtoolize.
- Allow override of PREFIX.
- Allow separate installation of utilities.
- Move quantize.c back into libgif.so to fix various library users.

--- Makefile.orig	2026-03-10 17:50:52.000000000 +0000
+++ Makefile
@@ -36,11 +36,11 @@ HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
 SOURCES = dgif_lib.c egif_lib.c gifalloc.c gif_err.c gif_font.c \
 	gif_hash.c openbsd-reallocarray.c quantize.c
 HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
-OBJECTS = $(SOURCES:.c=.o)
+OBJECTS = $(SOURCES:.c=.lo)
 
 USOURCES = qprintf.c getarg.c
 UHEADERS = getarg.h
-UOBJECTS = $(USOURCES:.c=.o)
+UOBJECTS = $(USOURCES:.c=.lo)
 
 UNAME:=$(shell uname)
 
@@ -75,7 +75,7 @@ UTILS = $(INSTALLABLE) \
 	gifwedge \
 	gif2rgb
 
-LDLIBS=libgif.a -lm
+LDLIBS=-lm
 
 MANUAL_PAGES_1 = \
 	doc/gifbuild.xml \
@@ -91,52 +91,37 @@ MANUAL_PAGES_7_MAN = $(MANUAL_PAGES_7:%.xml=%.7)
 MANUAL_PAGES_1_MAN = $(MANUAL_PAGES_1:%.xml=%.1)
 MANUAL_PAGES_7_MAN = $(MANUAL_PAGES_7:%.xml=%.7)
 
-SOEXTENSION	= so
+SOEXTENSION	= la
 LIBGIFSO	= libgif.$(SOEXTENSION)
 LIBGIFSOMAJOR	= libgif.$(SOEXTENSION).$(LIBMAJOR)
 LIBGIFSOVER	= libgif.$(SOEXTENSION).$(LIBVER)
 LIBUTILSO	= libutil.$(SOEXTENSION)
 LIBUTILSOMAJOR	= libutil.$(SOEXTENSION).$(LIBMAJOR)
-ifeq ($(UNAME), Darwin)
-SOEXTENSION	= dylib
-LIBGIFSO        = libgif.$(SOEXTENSION)
-LIBGIFSOMAJOR   = libgif.$(LIBMAJOR).$(SOEXTENSION)
-LIBGIFSOVER	= libgif.$(LIBVER).$(SOEXTENSION)
-LIBUTILSO	= libutil.$(SOEXTENSION)
-LIBUTILSOMAJOR	= libutil.$(LIBMAJOR).$(SOEXTENSION)
-endif
 
 SHARED_LIBS = $(LIBGIFSO) $(LIBUTILSO)
 STATIC_LIBS = libgif.a libutil.a
 
-all: shared-lib static-lib $(UTILS)
-ifeq ($(UNAME), Darwin)
-else
-	$(MAKE) -C doc
-endif
+all: $(LIBGIFSO) $(LIBUTILSO)
 
-$(UTILS):: $(STATIC_LIBS)
+.SUFFIXES: .lo
 
 shared-lib: $(SHARED_LIBS)
 
 static-lib: $(STATIC_LIBS)
 
-$(LIBGIFSO): $(OBJECTS) $(HEADERS)
-ifeq ($(UNAME), Darwin)
-	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o $(LIBGIFSO)
-else
-	$(CC) $(CFLAGS) $(CPPFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBGIFSOMAJOR) -o $(LIBGIFSO) $(OBJECTS)
-endif
+.c.lo:
+	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(CFLAGS) -c $<
 
-libgif.a: $(OBJECTS) $(HEADERS)
-	$(AR) rcs libgif.a $(OBJECTS)
+.libs/libgif.a: $(LIBGIFSO)
+.libs/libutil.a: $(LIBUTILSO)
 
+$(UTILS): .libs/libutil.a .libs/libgif.a
+
+$(LIBGIFSO): $(OBJECTS) $(HEADERS)
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(CFLAGS) $(LDFLAGS) -version-info $(LIBMAJOR):0:0 -rpath $(PREFIX)/lib -o $(LIBGIFSO) $(OBJECTS)
+
 $(LIBUTILSO): $(UOBJECTS) $(UHEADERS)
-ifeq ($(UNAME), Darwin)
-	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(UOBJECTS) -o $(LIBUTILSO)
-else
-	$(CC) $(CFLAGS) $(CPPLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS)
-endif
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(CFLAGS) $(LDFLAGS) -version-info $(LIBMAJOR):0:0 -rpath $(PREFIX)/lib -o $(LIBUTILSO) $(UOBJECTS)
 
 libutil.a: $(UOBJECTS) $(UHEADERS)
 	$(AR) rcs libutil.a $(UOBJECTS)
@@ -166,11 +151,7 @@ spellcheck:
 
 # Install/uninstall
 
-ifeq ($(UNAME), Darwin)
-install: all install-bin install-include install-lib
-else
 install: all install-bin install-include install-lib install-man install-doc
-endif
 
 install-bin: $(INSTALLABLE)
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
@@ -178,15 +159,9 @@ install-include:
 install-include:
 	$(INSTALL) -d "$(DESTDIR)$(INCDIR)"
 	$(INSTALL) -m 644 gif_lib.h "$(DESTDIR)$(INCDIR)"
-install-static-lib:
+install-lib:
 	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
-	$(INSTALL) -m 644 libgif.a "$(DESTDIR)$(LIBDIR)/libgif.a"
-install-shared-lib:
-	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
-	$(INSTALL) -m 755 $(LIBGIFSO) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOVER)"
-	ln -sf $(LIBGIFSOVER) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOMAJOR)"
-	ln -sf $(LIBGIFSOMAJOR) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSO)"
-install-lib: install-static-lib install-shared-lib
+	$(LIBTOOL) --mode=install --tag=CC $(BSD_INSTALL_LIB) $(LIBGIFSO) "$(DESTDIR)$(LIBDIR)"
 install-man:
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man1" "$(DESTDIR)$(MANDIR)/man7"
 	$(INSTALL) -m 644 $(MANUAL_PAGES_1_MAN) "$(DESTDIR)$(MANDIR)/man1"
